src/auth.rs

```diff
@@ -17,6 +21,53 @@ pub fn parse_auth(auth: String, host: &str) -> io::Result<(String, Option<String
     }
 }
 
+pub fn read_netrc() -> Option<String> {
+    let mut netrc_buf = String::new();
+    if let Some(mut hd_path) = home_dir() {
+        hd_path.push(".netrc");
+        if let Ok(mut netrc_file) = File::open(hd_path) {
+            if netrc_file.read_to_string(&mut netrc_buf).is_ok() {
+                return Some(netrc_buf);
+            };
+        };
+    };
+
+    None
+}
+
+pub fn auth_from_netrc(machine: &str, netrc: String) -> Option<(String, Option<String>)> {
+    if let Ok(netrc) = Netrc::parse_borrow(&netrc, false) {
+        let mut auths: Vec<(String, Option<String>)> = netrc
+            .machines
+            .iter()
+            .filter_map(|mach| {
+                if let Some(name) = &mach.name {
+                    if name.ends_with(machine) {
```

Comment:
Why `ends_with`? Shouldn't it be an exact match?

Reply:
Typically .netrc works on subdomains too, so `foo.org` works for `*.foo.org` but if your .netrc has `bar.foo.org` then it will only work on `*.bar.foo.org`.

But it might be better to be more explicit about this.

Reply:
HTTPie looks for an exact match, FWIW: [`netrc.py`](https://github.com/python/cpython/blob/374ee449331bc95d18c37f5032aaea1448462e58/Lib/netrc.py#L113)

(Which doesn't necessarily mean we should do the same, that implementation barely changed in two decades.)

Seems like `curl` doesn't match subdomains either, which software does?

If we do it, doing it right is a little tricky. `foo.org` should apply to `bar.foo.org`, but not to `barfoo.org`. And if the file lists `bar.foo.org` first and `foo.org` later then we would want to use the `bar.foo.org` configuration rather than `foo.org`.

I think I'd do a check like this:
```rust
match machine.strip_suffix(name) {
    Some(prefix) if prefix.is_empty() || prefix.ends_with('.') => {
        ...
    }
}
```
and then also compare to the length of the domain of the current best match to see if it's more or less specific.

Reply:
If both curl and httpie are explicit, then I think I’ll make it explicit as well. It might be my own fuzzy memory that’s at fault here.